react/components/Upload/Upload.tsx

```diff
@@ -0,0 +1,185 @@
+
+import StyleButton from './StyleButton'
+import UploadFileMutation from '../../graphql/uploadFile.graphql'
+import { FormRawInputProps, InputTypes } from '../../typings/InputProps'
+import { HiddenInput } from '../Input'
+
+interface MutationData {
+  uploadFile: { fileUrl: string }
+}
+
+const MAX_SIZE = 4 * 1024 * 1024
+
+const messages = defineMessages({
+  addBtn: {
+    defaultMessage: 'Add',
+    id: 'admin/form.add-button',
+  },
+  addTitle: {
+    defaultMessage: 'Insert document',
+    id: 'admin/form.add-document.title',
+  },
+  addLabel: {
+    defaultMessage: 'Document URL',
+    id: 'admin/form.add-document.label',
+  },
+  addPlaceholder: {
+    defaultMessage: 'URL',
+    id: 'admin/form.add-document.placeholder',
+  },
+  fileSizeError: {
+    defaultMessage:
+      'File exceeds the size limit of 4MB. Please choose a smaller one.',
+    id: 'admin/form.document-uploader.error.file-size',
+  },
+  genericError: {
+    defaultMessage: 'Something went wrong. Please try again.',
+    id: 'admin/form.document-uploader.error.generic',
+  },
+  uploadBtn: {
+    defaultMessage: 'Upload document',
+    id: 'admin/store.upload-button',
+  },
+  uploadLabel: {
+    defaultMessage: 'Choose an document to upload',
+    id: 'admin/store.upload-document-label',
+  },
+})
+
+const InputUpload = (props: FormRawInputProps) => {
+  const intl = useIntl()
+  const [uploadFile] = useMutation<MutationData>(UploadFileMutation)
+  const ref = React.useRef<HTMLDivElement>(null)
+  const [isLoading, setIsLoading] = React.useState(false)
+  const [fileName, setFileName] = React.useState('')
+  const [isOpen, setIsOpen] = React.useState(false)
+  const [imageUrl, setImageUrl] = React.useState<string | undefined>()
+  const [error, setError] = React.useState<string | null>()
+
+  const { inputType = InputTypes.input, pointer, ...rest } = props
+
+  const onDropImage = async (files: any[]) => {
+    setError(null)
+
+    try {
+      if (files?.[0]) {
+        setIsLoading(true)
+
+        const { data, errors } = await uploadFile({
+          variables: { file: files[0] },
+        })
+
+        if (errors) {
+          setError(intl.formatMessage(messages.fileSizeError))
+          return
+        }
+
+        setIsLoading(false)
+        setIsOpen(false)
+        setFileName(files?.[0].name)
+        setImageUrl(data?.uploadFile.fileUrl)
+
+        return data
+      }
+      return setError(intl.formatMessage(messages.fileSizeError))
+    } catch (err) {
+      setError(intl.formatMessage(messages.genericError))
+      setIsLoading(false)
+    }
+  }
+
+  const { getInputProps, getRootProps } = useDropzone({
+    accept: '.pdf, image/*',
+    maxSize: MAX_SIZE,
+    multiple: false,
+    onDrop: onDropImage,
+  })
+
+  const handleAddImage = () => {
+    setIsOpen(false)
+  }
+
+  return (
+    <div className="vtex-styleguide-9-x-dropdown vtex-dropdown" ref={ref}>
+      <span className="db mb3 w-100 c-on-base t-small">
+        Operating Agreement
```

Comment:
Shouldn't internationalize this message?

Reply:
Done!

Reply:
@santosluiz000 you've added the definitions but you also need to add `IOMessage` here so it can internationalize correctly.

You should also use `defineMessages` function to declare all i18n strings you use inside react. Take a look [here](https://github.com/vtex-apps/store-form/blob/f42b4e84aa7d02f63f31a55249b173fba5106ee0/react/FormSubmit.tsx).

Reply:
This is still here @santosluiz000, this message will not be internationalized if it stays like this.